core/src/database/watchtower.rs

```diff
@@ -0,0 +1,273 @@
+//! # Watchtower Related Database Operations
+//!
+//! This module includes database functions which are mainly used by a watchtower.
+
+use super::Database;
+use crate::errors::BridgeError;
+use crate::execute_query_with_tx;
+use bitcoin::{ScriptBuf, XOnlyPublicKey};
+use bitvm::signatures::winternitz;
+use bitvm::signatures::winternitz::PublicKey as WinternitzPublicKey;
+use sqlx::Postgres;
+
+impl Database {
+    /// Sets Winternitz public keys for an operator.
+    pub async fn save_watchtower_winternitz_public_keys(
```

Comment:
Can we clarify the action of the function? "save" is ambiguously used for both insert and upsert (ie. insert if not exists else update). This one is an insert, so rename to `insert_watchtower_winternitz_public_keys`.